File: api/src/models/Company.js
import mongoose from 'mongoose';

const companySchema = new mongoose.Schema(
  {
    code: {
      type: String,
      required: true,
      unique: true,
      trim: true,
      uppercase: true
    },
    businessName: {
      type: String,
      required: true,
      trim: true
    },
    address: {
      type: String,
      trim: true,
      default: ''
    },
    phone: {
      type: String,
      trim: true,
      default: ''
    },
    email: {
      type: String,
      trim: true,
      default: ''
    },
    gstin: {
      type: String,
      trim: true,
      default: ''
    },
    state: {
      type: String,
      trim: true,
      default: ''
    },
    placeOfSupply: {
      type: String,
      trim: true,
      default: ''
    },
    bankName: {
      type: String,
      trim: true,
      default: ''
    },
    bankAccountNo: {
      type: String,
      trim: true,
      default: ''
    },
    bankIfsc: {
      type: String,
      trim: true,
      default: ''
    },
    bankAccountHolder: {
      type: String,
      trim: true,
      default: ''
    },
    logoText: {
      type: String,
      trim: true,
      default: ''
    },
    /** Public API path or legacy inline data URI (migrated to disk on read). */
    logoImage: {
      type: String,
      default: ''
    },
    terms: {
      type: [String],
      default: []
    },
    isDefault: {
      type: Boolean,
      default: false
    },
    activeStatus: {
      type: Boolean,
      default: true
    }
  },
  { timestamps: true }
);

companySchema.index({ businessName: 'text', code: 'text', gstin: 'text' });
companySchema.index({ isDefault: 1 });

export const Company = mongoose.model('Company', companySchema);

File: ims-api-net/src/Ims.Domain/Masters/Company.cs
using Ims.Domain.Common;

namespace Ims.Domain.Masters;

public sealed class Company : EntityBase, IYearScoped
{
    public string YearDatabaseName { get; set; } = string.Empty;

    public string Code { get; set; } = string.Empty;

    public string BusinessName { get; set; } = string.Empty;

    public string? Tagline { get; set; }

    public string? Address { get; set; }

    public string? Phone { get; set; }

    public string? Email { get; set; }

    public string? Gstin { get; set; }

    public string? State { get; set; }

    public string? PlaceOfSupply { get; set; }

    public string? LogoText { get; set; }

    public string? LogoImage { get; set; }

    public bool IsDefault { get; set; }

    public bool ActiveStatus { get; set; } = true;
}
